addons/loyalty_delivery/tests/test_loyalty_delivery.py

```diff
@@ -17,28 +18,28 @@ def setUpClass(cls):
         })
         cls.product_4 = cls.env['product.product'].create({'name': 'A product to deliver'})
         cls.product_uom_unit = cls.env.ref('uom.product_uom_unit')
-
-    def test_delivery_cost_gift_card(self):
-        """
-        A customer has a carrier with the amount greater than the one to have
-        free shipping cost, then uses a gift card that lowers that amount to less
-        than the threshold: the shipping cost should still be 0.0
-        """
-
-        product_delivery_free = self.env['product.product'].create({
+        cls.product_delivery_free = cls.env['product.product'].create({
             'name': 'Free Delivery Charges',
             'type': 'service',
             'list_price': 40.0,
-            'categ_id': self.env.ref('delivery.product_category_deliveries').id,
+            'categ_id': cls.env.ref('delivery.product_category_deliveries').id,
         })
-        free_delivery = self.env['delivery.carrier'].create({
+        cls.free_delivery = cls.env['delivery.carrier'].create({
             'name': 'Delivery Now Free Over 100',
             'fixed_price': 40,
             'delivery_type': 'fixed',
-            'product_id': product_delivery_free.id,
+            'product_id': cls.product_delivery_free.id,
             'free_over': True,
             'amount': 100,
         })
+
+    def test_delivery_cost_gift_card(self):
+        """
+        A customer has a carrier with the amount greater than the one to have
+        free shipping cost, then uses a gift card that lowers that amount to less
+        than the threshold: the shipping cost should still be 0.0
+        """
```

Comment:
That's not the case in fact. Here are the so line you get:
- PC Assamble: price total of 138
- Gift Card: price total of -4
- Free Shipping: price total of 0

==> You are not testing the carrier is free Over 100 even with a gift card that lower the price under 100. PC + Gift card = 134, still above 100